### DISCUSSION TITLE
Scaling for initial condition, while using log molar form?

### DISCUSSION THREAD

**Discussion by ananthanarasimhanj** [shannon-lab/zapdos#245] [Q&A Plasma Chemistry / CRANE]
Is there a way to use scaling for initial condition, while using log molar form?

I am already using `automatic_scaling = true` in the `[Executioner]` block; whether this automatic scaling is for time step, or does it work for initial conditions? 

The initial condition I use for N2 and H2O species is usually of value between 50 and 60 (log molar form). But when N2 is in 50-60, and H2O is <50, I find issues in solving the chemistry. I am thinking if I need to do any manual scaling ( I don't have the working knowledge of `automatic_scaling = true`.

**Comment by csdechant** [comment]
You can do manual scaling of a single variable within the `Variables` block. Below is an example:

` [e]`
`    family = SCALAR`
`    order = FIRST`
`    initial_condition = 13.815510557964274`
`    scaling = 0.1`
` []`

Though I should note that if you are using log molar form and `automatic_scaling = true`, it seems unlikely scaling is the issue (though not impossible). For more information about MOOSE auto scaling, please follow this [link](https://mooseframework.inl.gov/source/systems/NonlinearSystemBase.html).

**Reply by ananthanarasimhanj** [reply]
Thanks! I will try this. Whether the scaline is a multiplication factor?  What is the implication of 0.1 on the value 13.8155? Does `scaling = 0.1` makes that make `initial_condition ` as 1 (close to unity)? So, If I need to scale H2O of vale 40 to match with N2 of value 50-60, should I use the scaling factor on H2O or N2?

**Reply by csdechant** [reply]
Yes, the `scaling` parameter is a multiplication factor. In particular, it applies a one time scaling factor during the initial setup for the specified variable during the matrix solves. The main idea behind any matrix scaling is to have the variable entries be close in magnitude and ideally near unity.

As for the implication of 0.1 on the value 13.8155 for the example, the initial condition is still 13.8155. During the matrix solve, the matrix entry is 1.38155, since all entries involving `e` are scaled by 0.1.

As to the question of whether to scale H2O or N2, I don't believe I can give you a definitive answer. From a debugging standpoint, try scaling H2O first and see what happens.

**Reply by ananthanarasimhanj** [reply]
Thanks! I am using `compute_scaling_once = false` in the Executioner block along with `automatic_scaling = true`. Is that OK ?

**Reply by csdechant** [reply]
Yes, if you believe that the magnitude of your variables will changes drastically during a transient solve (which can be the case for plasma chemistry). The only major downside is that it might slow down your solve a bit because MOOSE is computing a new automatic scaling term each time step, but I never notice a big slow down when I use `compute_scaling_once = false`.

**Reply by ananthanarasimhanj** [reply]
Thanks! I am wondering what else could be the issue, if automatic_scaling is taking care of the initial condition of the species density for the matrix. I guess I should probably investigate the reaction mechanism.
